cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -216,6 +224,14 @@ static void sc_populate_libgl_with_hostfs_symlinks(const char *libgl_dir,
 				 "%s/%s", libgl_dir, filename);
 		debug("creating symbolic link %s -> %s", symlink_name,
 		      symlink_target);
+
+		// Make sure we don't have some link already (merged GLVND systems)
+		if (lstat(symlink_name, &stat_buf) == 0) {
+			if (unlink(symlink_name) != 0) {
```

Comment:
Shouldn't you also be testing if S_IFLNK? lstat() will happily return 0 if regular file or symlink.

Reply:
The path is inside our hostfs symlink farm that we create, and we only create symlinks, the code doesn't copy files.

Reply:
Note, the naming here is confusing, but `symlink_name` is **not** the source name, it is the **target** name for the symlink, i.e. the new inode. Confused the hell out of me going through it the first time.

Reply:
I'll probably do a small update here to use consistent terminology (oldname vs whatever the other one was).